mapreduce/datadog_checks/mapreduce/mapreduce.py

```diff
@@ -410,38 +421,29 @@ def _rest_request_to_json(self, address, object_path, service_name, tags=None, *
             response_json = response.json()
 
         except Timeout as e:
-            self.service_check(
-                service_name,
-                AgentCheck.CRITICAL,
-                tags=service_check_tags,
-                message="Request timeout: {}, {}".format(url, e),
-            )
+            self._critical_service(service_name, service_check_tags, "Request timeout: {}, {}".format(url, e))
             raise
 
         except (HTTPError, InvalidURL, ConnectionError) as e:
-            self.service_check(
-                service_name,
-                AgentCheck.CRITICAL,
-                tags=service_check_tags,
-                message="Request failed: {}, {}".format(url, e),
-            )
+            self._critical_service(service_name, service_check_tags, "Request failed: {}, {}".format(url, e))
             raise
 
         except JSONDecodeError as e:
-            self.service_check(
-                service_name,
-                AgentCheck.CRITICAL,
-                tags=service_check_tags,
-                message="JSON Parse failed: {}, {}".format(url, e),
-            )
+            self._critical_service(service_name, service_check_tags, "JSON Parse failed: {}, {}".format(url, e))
             raise
 
         except ValueError as e:
-            self.service_check(service_name, AgentCheck.CRITICAL, tags=service_check_tags, message=str(e))
+            self._critical_service(service_name, service_check_tags, str(e))
             raise
 
         return response_json
 
+    def _critical_service(self, service_name, tags, message):
+        if service_name:
+            self.service_check(
+                service_name, AgentCheck.CRITICAL, tags=tags, message=message,
+            )
```

Comment:
Why the `if service_name`, I don't see case where `service_name` is falsy.

Reply:
Line 202: `cluster_info = self._rest_request_to_json(self.rm_address, self.CLUSTER_INFO)` no service name is provided